python/ray/workflow/http_event_provider.py

```diff
@@ -0,0 +1,261 @@
+    and returns a handle to it.
+    """
+    try:
+        return ray.serve.get_deployment(common.HTTP_EVENT_PROVIDER_NAME).get_handle(sync=True)
+    except:
+        # HTTPEventProvider Serve endpoint was not found. Create an deployment.
+        HTTPEventProvider.deploy()
+        return ray.serve.get_deployment(common.HTTP_EVENT_PROVIDER_NAME).get_handle(sync=True)
+
+class HTTPListener(EventListener):
+    """HTTPLister is defined to work with the HTTPEventProvider. It implements two
+    APIs, poll_for_event() and event_checkpointed(). An instance of HTTPListener will
+    be started by the get_message() of the workflow.wait_for_event() to listen for an event
+    from the HTTPEventProvider instance (a Ray Serve deployment). Another instance of HTTPListener
+    will be started by the message_committed() of the workflow.wait_for_event() to confirm
+    that the event has been checkpointed.
+
+
+    Example definition
+    ==================
+
+    ```
+    class HTTPListener:
+
+        def __init__(self):
+            self.handle = get_or_create_http_event_provider()
+
+        async def poll_for_event(self, event_key) -> Event:
+
+        async def event_checkpointed(self, event) -> None:
+
+    ```
+
+    Example Usage
+    =============
+    >>> from ray.workflow.http_event_provider import HTTPEventProvider, HTTPListener
+    >>> ray.init(address='auto', namespace='serve')
+    >>> serve.start(detached=True)
+    >>> event_node = workflow.wait_for_event( # doctest: +SKIP
+    ...     HTTPListener, event_key='')
+    >>> handle_event = ... # doctest: +SKIP
+    >>> handle_event.bind(event_node).run() # doctest: +SKIP
+    >>>
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.handle = get_or_create_http_event_provider()
+
+    async def poll_for_event(self, event_key: str = None) -> Event:
+        """workflow.wait_for_event calls this method to subscribe to the HTTPEventProvider
+        and return the received external event
+        Args:
+            event_key: a unique identifier to the receiving node in a workflow; if missing,
+            default to current workflow step id
+        Returns:
+            tuple(event_key, event_payload)
+        """
+        workflow_id = workflow_context.get_current_workflow_id()
+        if event_key is None:
+            event_key = workflow_context.get_current_step_id()
+
+        event_key_payload = await self.handle.get_event_payload.remote(workflow_id, event_key)
+        return event_key_payload
+
+    async def event_checkpointed(self, event: Event) -> None:
+        """workflow.wait_for_event calls this method after the event has been checkpointed
+        and a transaction can be safely committed."""
+        (event_key, _) = event
+        resp = await self.handle.report_checkpointed.remote(
+            workflow_context.get_current_workflow_id(), event_key, True)
+        if resp != "OK":
```

Comment:
We don't need to check this. Just try catch is good enough.

Reply:
try catch in `HTTPEventProvider.report_checkpointed`